openreview/profile/management.py

```diff
@@ -28,75 +20,252 @@ def setup(self):
+                                'value': {
+                                    'param': {
+                                        'type': 'string[]',
+                                        'optional': True
+                                    }
+                                }
+                            },                        
+                            'venue': {
+                                'order': 3,
+                                'description': 'Enter the venue where the paper was published.',
+                                'value': {
+                                    'param': {
+                                        'type': 'string'
+                                    }
+                                }
+                            },
+                            'venueid': {
+                                'order': 4,
+                                'value': {
+                                    'param': {
+                                        'type': "string",
+                                        'const': dblp_group_id,
+                                    }
+                                }
+                            }
+                        }
+                    }                                        
                 }
-            }
-        ))                           
+            )
+        )
+
+        author_coreference_invitation_id = f'{dblp_group_id}/-/Author_Coreference'
+
+        with open(os.path.join(os.path.dirname(__file__), 'process/dblp_author_coreference_pre_process.js'), 'r') as f:
+            file_content = f.read()
+
+        self.client.post_invitation_edit(
+            invitations = meta_invitation_id,
+            signatures = [dblp_group_id],
+            invitation = openreview.api.Invitation(
+                id=author_coreference_invitation_id,
+                readers=['everyone'],
+                writers=[dblp_group_id],
+                signatures=[dblp_group_id],
+                invitees=['~'],
+                preprocess=file_content,
+                edit={
+                    'readers': ['everyone'],
+                    'signatures': { 
+                        'param': { 
+                            'items': [
+                                { 'prefix': '~.*', 'optional': True },
+                                { 'value': self.support_group_id, 'optional': True },
+                                { 'value': dblp_uploader_group_id, 'optional': True } 
+                            ]
+                        } 
+                    },
+                    'writers':  [dblp_group_id, '${2/signatures}'],
+                    'content': {
+                        'author_index': {
+                            'value': {
+                                'param': {
+                                    'type': 'integer'
+                                }
+                            }
+                        },
+                        'author_id' : {
```

Comment:
i think user may not be able to find this author id if it's not their preferred name

Reply:
I was thinking about using the signature as author_id but this means this invitation can only be used by the author of the paper.

Reply:
for example the user has 2 names ABC and XYZ where ABC is the preferred name so when user go to profile page, the url show id=~ABC1

when the name XYZ is used in the DBLP publication, the user need to put the id ~XYZ1 in this field but currently the user can't find this id anywhere unless changing the preferred name to XYZ (and go to profile page) or query api directly and check the username of the name XYZ

Reply:
I think the preprocess validation lets the user use any username, it doesn't need to match with the name listed in the DBLP publication.

Reply:
i think the preprocess function fails if the name does not match
invitation validation also fails when author id is changed by editing the author coreference edit